### ISSUE TITLE
[3.2.0] null_resource id becomes constant, semi-random and unaffected by changing of triggers values

### ISSUE THREAD

**Issue by agrzegorczyk-leonsoftware** [hashicorp/terraform-provider-null#159]
### Terraform CLI and Provider Versions

Terraform v1.2.3
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v4.39.0
+ provider registry.terraform.io/hashicorp/external v2.2.3
+ provider registry.terraform.io/hashicorp/null v3.2.0

### Terraform Configuration

```terraform
resource "null_resource" "build-trigger" {
  triggers = {
    test-trigger-value = "qwertyuiop12345678"
    source-image-id   = data.aws_ami.latest-amazon-ecs-optimized.id
    assets-dir-hash   = data.external.assets-dir-hash.result["sha256"]
    packer-definition = filesha256("./packer.toml")
  }
}
```

### Expected Behavior

null_resource.[resource-name].id  is random and changes varies with triggers, as before 3.2.0 version of null provider.

### Actual Behavior

null_resource.[resource-name].id value is constant (5577006791947779410 for first null_resource instance in module ,as I see), and doesn't change after changed triggers value or recreate of resource.

### Steps to Reproduce

Change any null_resource trigger value and verify that the resource ID stays the same after apply.

### How much impact is this issue causing?

Low

### Additional Information

I see this as a backward compatibility breaking change, but deployed in minor provider update.
In my case this breaks unique resource naming, which depends on triggers value.
I realize there is a way around this with a random_id from a random provider and that will be my solution, but this "quiet" change has caused some hassle and troubleshooting.

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by agrzegorczyk-leonsoftware** [comment]
Aditionally during plan, if I change triggers, id of null_resource resource is marked as "(known after apply)", despite it never changes, and this causes unnecessary recreation of dependent resources.

**Comment by SBGoods** [comment]
Hi @agrzegorczyk-leonsoftware, thank you for raising this issue and sorry that you've been running into trouble here. This is not the intended behavior for the id attribute and I've introduced a fix for this in #160 that will be included in the next release. 

Thank you again and please reach out with any questions!

**Comment by agrzegorczyk-leonsoftware** [comment]
Hi @SBGoods,
thank you very much for quick response and remediation of this issue.
I wasn't sure if this was a bug or intended new behavior.
